pkg/agent/sysadvisor/plugin/poweraware/evictor/server/evictor_service.go

```diff
@@ -0,0 +1,181 @@
+}
+
+// GetEvictPods is called from a remote evict plugin client to get evict candidates
+// In the current eviction manager framework, plugins are expected to implement either GetEvictPods or GetTopEvictionPods + ThresholdMet;
+// the former allows the plugin to explicitly specify force and soft eviction candidates, which suits this plugin's use case.
+// Adequate to implement only GetEvictPods and simply let GetTopEvictionPods and ThresholdMet return default responses.
+func (p *powerPressureEvictServer) GetEvictPods(ctx context.Context, request *pluginapi.GetEvictPodsRequest) (*pluginapi.GetEvictPodsResponse, error) {
+	general.InfofV(6, "pap: evict: GetEvictPods request with %d active pods", len(request.GetActivePods()))
+	activePods := map[types.UID]struct{}{}
+	for _, pod := range request.GetActivePods() {
+		if len(pod.GetUID()) > 0 { // just in case of invalid input
+			activePods[pod.GetUID()] = struct{}{}
+		}
+	}
+
+	p.mutex.RLock()
+	defer p.mutex.RUnlock()
+
+	evictPods := make([]*pluginapi.EvictPod, 0)
+
+	pods := p.evicts
+	for _, v := range pods {
+		if _, ok := activePods[v.GetUID()]; ok {
+			evictPods = append(evictPods, &pluginapi.EvictPod{
+				Pod:                v,
+				Reason:             evictReason,
+				ForceEvict:         true,
+				EvictionPluginName: EvictionPluginNameNodePowerPressure,
+			})
+		}
+	}
+
+	general.InfofV(6, "pap: evict: GetEvictPods respond with %d pods to evict", len(evictPods))
+	return &pluginapi.GetEvictPodsResponse{EvictPods: evictPods}, nil
+}
+
+func newPowerPressureEvictServer() *powerPressureEvictServer {
+	return &powerPressureEvictServer{
+		evicts: make(map[types.UID]*v1.Pod),
+	}
+}
+
+func newPowerPressureEvictService(conf *config.Configuration, emitter metrics.MetricEmitter) (evictor.PodEvictor, error) {
+	plugin := newPowerPressureEvictServer()
+	regWrapper, err := skeleton.NewRegistrationPluginWrapper(plugin,
+		[]string{conf.PluginRegistrationDir}, // unix socket dirs
+		func(key string, value int64) {
+			_ = emitter.StoreInt64(key, value, metrics.MetricTypeNameCount, metrics.ConvertMapToTags(map[string]string{
+				"pluginName": EvictionPluginNameNodePowerPressure,
+				"pluginType": registration.EvictionPlugin,
+			})...)
+		})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to register pap power pressure eviction service")
+	}
+
+	plugin.service = regWrapper
+	return plugin, nil
+}
+
+func NewPowerPressureEvictionPlugin(conf *config.Configuration, emitter metrics.MetricEmitter) (podEvictor evictor.PodEvictor, err error) {
+	return startPowerPressurePodEvictorService(conf, emitter)
+}
+
+func startPowerPressurePodEvictorService(conf *config.Configuration, emitter metrics.MetricEmitter) (evictor.PodEvictor, error) {
+	podEvictor, err := newPowerPressureEvictService(conf, emitter)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create power pressure eviction plugin server")
+	}
+
+	return podEvictor, nil
+}
```

Comment:
Now that we don't start the server within these functions, the function name should be changed. Also, is it possible to name these functions using fewer concepts? It's hard to tell the semantic difference between `server`, `service`, and `plugin`.

Reply:
good catch! will consolidate these 3 functions into one